impeller/renderer/backend/metal/pipeline_library_mtl.mm

```diff
@@ -59,7 +52,31 @@
   descriptor.stencilAttachmentPixelFormat =
       ToMTLPixelFormat(desc.GetStencilPixelFormat());
 
-  return descriptor;
+  const auto& constants = desc.GetSpecializationConstants();
```

Comment:
This is still a bit iffy, but we improve performance by not blocking on the specialization, it all goes into the same pipeline future which is only blocking once we actually trry to use it.

Since the compilation is out of process anyway, blocking on it is pointless and only serves to park the thread.

Reply:
Feels like we're writing javascript here with this chain of nested completion callbacks. ;) Looks safe to me, though.